ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc

```diff
@@ -412,128 +374,138 @@ TFCore::getOutputTensorDim (GstTensorsInfo * info)
-  bool OwnsMemory () const override { return false; }
-};
+static void
+DeallocateTensor (void* data, std::size_t, void*) {
+  /* do nothing, the data will be free at the last of pipeline */
+  return;
+}
 
 /**
  * @brief	run the model with the input.
  * @param[in] input : The array of input tensors
  * @param[out]  output : The array of output tensors
  * @return 0 if OK. non-zero if error.
- *         -1 if the model does not work properly.
- *         -2 if the output properties are different with model.
+ *        -1 if encoding STRING is failed.
+ *        -2 if running session is failed.
  */
 int
 TFCore::run (const GstTensorMemory * input, GstTensorMemory * output)
 {
 #if (DBG)
   gint64 start_time = g_get_real_time ();
 #endif
-
-  std::vector <std::pair <string, Tensor>> input_feeds;
-  std::vector <Tensor> outputs;
-
-  for (int i = 0; i < inputTensorMeta.num_tensors; ++i) {
-    Tensor in;
-
-    /* If the datatype is STRING, it should be handled in specific process */
-    if (input_tensor_info[i].type == DT_STRING) {
-      in = Tensor (input_tensor_info[i].type, input_tensor_info[i].shape);
-      in.scalar<string>()() = string ((char *) input[i].data, input[i].size);
-    } else {
-      if (mem_optmz) {
-        TFBuffer *buf;
-        TF_DataType dataType;
-
-        if (!getTensorTypeToTF_Capi (input[i].type, &dataType)){
-          g_critical ("This data type is not valid: %d", input[i].type);
-          return -1;
-        }
-
-        /* this input tensor should be UNREF */
-        buf = new TFBuffer (input[i].data);
-        buf->len_ = input[i].size;
-
-        in = TensorCApi::MakeTensor (
-          dataType,
-          input_tensor_info[i].shape,
-          buf
-        );
-
-        buf->Unref();
-
-        if (!in.IsAligned ()) {
-          g_critical ("the input tensor %s is not aligned", inputTensorMeta.info[i].name);
-          return -2;
-        }
-      } else {
-        in = Tensor (input_tensor_info[i].type, input_tensor_info[i].shape);
-        /* copy data */
-        std::copy_n ((char *) input[i].data, input[i].size,
-            const_cast<char *>(in.tensor_data().data()));
+  std::vector<TF_Output> input_ops;
+  std::vector<TF_Tensor*> input_tensors;
+  std::vector<TF_Output> output_ops;
+  std::vector<TF_Tensor*> output_tensors;
+  TF_Status* status = TF_NewStatus ();
+  char *input_encoded = nullptr;
+
+  // create input tensor for the graph from `input`
+  for (int i = 0; i < inputTensorMeta.num_tensors; i++) {
+    TF_Tensor* in_tensor = nullptr;
+    TF_Output input_op = {
+      TF_GraphOperationByName (graph, inputTensorMeta.info[i].name), 0
+      };
+    g_assert (input_op.oper != nullptr);
+    input_ops.push_back (input_op);
+
+    if (input_tensor_info[i].type == TF_STRING){
+      size_t encoded_size = TF_StringEncodedSize (input[i].size);
+      size_t total_size = 8 + encoded_size;
+      input_encoded = (char*) malloc (total_size);
+      for (int i =0; i < 8; ++i) {
```

Comment:
line 407 (for int i=0;..)
and this line uses i also (change to another name)

Reply:
it's fixed!